### ISSUE TITLE
[GCC11] HeterogeneousCore/CUDAUtilities: array subscript is partly outside array bounds

### ISSUE THREAD

**Issue by iarspider** [cms-sw/cmssw#34917]
Build of HeterogeneousCore/CUDAUtilities fails with GCC11:

```
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/e9c933c3dc3076293cddde4dd702cd2f/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-08-16-1100/src/HeterogeneousCore/CUDAUtilities/test/radixSort_t.cu: In function 'void go(bool) [with T = float; int NS = 2; U = float; LL = double]':
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/e9c933c3dc3076293cddde4dd702cd2f/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-08-16-1100/src/HeterogeneousCore/CUDAUtilities/test/radixSort_t.cu:145:15: error: array subscript 'long unsigned int[0]' is partly outside array bounds of 'float [1]' [-Werror=array-bounds]
   145 |           *k = (*k >> sh) << sh;
      |           ~~~~^~~~~~~~
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/e9c933c3dc3076293cddde4dd702cd2f/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-08-16-1100/src/HeterogeneousCore/CUDAUtilities/test/radixSort_t.cu:139:6: note: while referencing 'k1'
  139 |         auto k1 = a[ind[j]];
      |      ^ 
```

Log file: https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc11/CMSSW_12_1_X_2021-08-16-1100/HeterogeneousCore/CUDAUtilities

**Comment by iarspider** [comment]
assign heterogeneous

**Comment by makortel** [comment]
@Vininn The error points to the same helper function we discussed in https://github.com/cms-patatrack/pixeltrack-standalone/pull/195#discussion_r677882403 .

**Comment by fwyzard** [comment]
See #34929 for a possible fix.

**Comment by smuzaffar** [comment]
this has been fixed via  #34929
